examples/application/python_editor/python_editor_task.py

```diff
@@ -333,15 +333,15 @@ def _prompt_for_save(self):
 
     # Trait change handlers --------------------------------------------------
 
-    @on_trait_change("window:closing")
+    @observe("window:closing")
     def _prompt_on_close(self, event):
         """ Prompt the user to save when exiting.
         """
         close = self._prompt_for_save()
-        event.veto = not close
+        event.new.veto = not close
```

Comment:
assuming that `event.new` is `window`

```suggestion
        window = event.new
        window.veto = not close
```